src/mon/MgrMap.h

```diff
@@ -234,9 +322,18 @@ class MgrMap
     m.print_summary(nullptr, &ss);
     return out << ss.str();
   }
+
+  friend ostream& operator<<(ostream& out, const std::vector<ModuleInfo>& mi) {
+    for (const auto &i : mi) {
```

Comment:
nit, could use
```c++
std::copy(begin(mi), end(mi), ceph::make_ostream_joiner(out, ' '));
```

to avoid the trailing space in the output.

Reply:
Because it's an array of ModuleInfo (not strings), it's kind of awkward to do something like that (iirc this method is only used in debug output)

Reply:
ahh, right!